source/docs/software/basic-programming/coordinate-system.rst

```diff
@@ -0,0 +1,325 @@
+
+Historically, FRC has used two types of field layouts in relation to the red and blue alliance.
+
+Games such as Rapid React in 2022 used a rotated layout. A rotated layout means that, from your perspective from behind your alliance wall, your field elements and your opponents elements are in the same location. Notice in the Rapid React field layout diagram below, whether you are on the red or blue alliance, your human player station is on your right and your hanger is on your left.
+
+.. figure:: images/coordinate-system/rapid-react-field.jpg
+   :alt: Rotated Rapid React field from 2022
+
+   Rotated field from RAPID REACT in 2022 [#]_
+
+Games such as CHARGED UP in 2023 and CRESCENDO in 2024 used a mirrored layout. A mirrored layout means that the red and blue alliance layout are mirrored across the centerpoint of the field. Refer to the CHARGED UP field diagram below. When you are standing behind the blue alliance wall, the charge station is on the right side of the field from your perspective. However, standing behind the red alliance wall, the charge station is on the left side of the field from your perspective.
+
+.. figure:: images/coordinate-system/charged-up-field.jpg
+   :alt: Mirrored CHARGED UP field from 2023
+
+   Mirrored field from Charged Up in 2023 [#]_
+
+Dealing with red or blue alliance
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There are two primary ways many teams choose to define the field coordinate system. In both methods, positive rotation (theta) is in the counter-clockwise (CCW) direction.
+
+Always blue origin
+~~~~~~~~~~~~~~~~~~
+
+You may choose to define the origin of the field on the blue side, and keep it there regardless of your alliance color. With this solution, positive x-axis points away from the blue alliance wall.
+
+.. figure:: images/coordinate-system/field-blue-alliance.svg
+   :alt: CHARGED UP with blue origin
+   :scale: 200
+
+   CHARGED UP with blue origin
+
+Some advantages to this approach are:
+
+- Pose estimation with AprilTags is simplified. AprilTags throughout the field are unique. If you keep the coordinate system the same regardless of alliance, there is no need for special logic to deal the location of AprilTags on the field relative to your alliance.
+- Many of the tools and libraries used in FRC follow this convention. Some of the tools include: PathPlanner, Choreo, and the ShuffleBoard Field2d widget.
+
+In order to use this approach for field oriented driving, driver input needs to consider the alliance color. When your alliance is red and the driver is standing behind the red alliance wall, they will want the robot to move toward down field toward the blue alliance wall. However, when your alliance is blue, the driver will want the robot to go down field toward the red alliance wall.
+
+A simple way to deal with field oriented driving is to check the alliance color reported by the `DriverStation` class, and invert the drivers controls based on the alliance.
```

Comment:
This probably needs a callout about how to set the driver station color in the driver station for practice. And also a warning about not checking once in robotInit. They are down below, but really apply to both approaches, right?

Reply:
Good ideas. I'll add info about changing the alliance in Driver Station for practice.
With the Always Blue approach, the Driver Station is checked every iteration and an alliance change is handled automatically.
I didn't go into detail on how to handle an alliance change with for the other approach, because it's complicated and varied by code base.
Maybe I can add some detail about about that, I'll have to give it some thought.

Reply:
It was implied that it's checked every iteration because it was in the same code block, but it wasn't stated unless I missed it. My worry is that someone would see that and think they could move it operate once.

Reply:
Oh, I see what you mean. The snippet doesn't have context that it's in a block that gets called every iteration, like `execute()` , so it's not clear that `invert` is being set every iteration. I'll find a way to make that more clear.